src/pages/verify-email.tsx

```diff
@@ -0,0 +1,180 @@
+import {
+  Alert,
+  Button,
+  Card,
+  Center,
+  Container,
+  Group,
+  Loader,
+  Stack,
+  Text,
+  Title,
+} from '@mantine/core';
+import { IconCheck, IconX } from '@tabler/icons-react';
+import type { GetServerSideProps } from 'next';
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+import { Meta } from '~/components/Meta/Meta';
+import { trpc } from '~/utils/trpc';
+
+type VerifyEmailPageProps = {
+  token?: string;
+};
+
+export default function VerifyEmailPage({ token }: VerifyEmailPageProps) {
+  const router = useRouter();
+  const [status, setStatus] = useState<'pending' | 'loading' | 'success' | 'error'>('pending');
+  const [message, setMessage] = useState('');
+  const [newEmail, setNewEmail] = useState('');
+  const [currentEmail, setCurrentEmail] = useState('');
+
+  const { data: tokenData, error: tokenError } = trpc.user.validateEmailToken.useQuery(
+    { token: token || '' },
+    { enabled: !!token }
+  );
+
+  const verifyEmailMutation = trpc.user.verifyEmailChange.useMutation({
+    onSuccess: (data) => {
+      setStatus('success');
+      setMessage(data.message);
+    },
+    onError: (error) => {
+      setStatus('error');
+      setMessage(error.message);
+    },
+  });
+
+  // Update status based on token validation
+  useEffect(() => {
+    if (!token) {
+      setStatus('error');
+      setMessage('No verification token provided');
+    } else if (tokenError) {
+      setStatus('error');
+      setMessage(tokenError.message);
+    } else if (tokenData) {
+      setStatus('pending');
+      setNewEmail(tokenData.newEmail);
+      setCurrentEmail(tokenData.currentEmail);
+    }
+  }, [token, tokenData, tokenError]);
+
+  const handleConfirmChange = () => {
+    if (token) {
+      setStatus('loading');
+      verifyEmailMutation.mutate({ token });
+    }
+  };
+
+  const handleReturnToAccount = () => {
+    router.push('/user/account');
+  };
+
+  return (
+    <>
+      <Meta title="Email Verification - Civitai" />
```

Comment:
Might be a good idea to prevent indexing this page 🤷 
```suggestion
      <Meta title="Email Verification - Civitai" deIndex />
```